internal/session/session.go

```diff
@@ -367,7 +377,7 @@ func contains(ss []string, t string) bool {
 }
 
 // newCert creates a new session certificate. If addresses are supplied, they will be parsed and added to IP or DNS SANs as appropriate.
-func newCert(ctx context.Context, wrapper wrapping.Wrapper, userId, jobId string, addresses []string, exp time.Time) (ed25519.PrivateKey, []byte, error) {
+func newCert(ctx context.Context, wrapper wrapping.Wrapper, userId, jobId string, addresses []string, exp time.Time, rand io.Reader) (ed25519.PrivateKey, []byte, error) {
```

Comment:
perhaps test rand for IsNil(...) or are you relying on ed25519.GenerateKey(...) to handle nil properly?

Reply:
perhaps test rand for IsNil(...)?  I'm not sure if CreateCertificate(...) does the right thing.

Reply:
I looked at this method with a fresh set of eyes and realised I could remove half the parameters, so now it's both tested and much slimmer. Good shout!

Reply:
Much better!!!!!